From e712b8f24d924b0f0f027a5cfd9daf5e75eeac5b Mon Sep 17 00:00:00 2001
From: Arne Coucheron <arco68@gmail.com>
Date: Wed, 4 Sep 2013 04:46:42 +0200
Subject: [PATCH] libstagefright: Add support for custom LPA buffer size in
 legacy LPAPlayer

     - on msm7x27a with bootloader heap for lpa audio during audio playback with,
       DSP Manager effects enabled there is a lot of stuttering with the default buffer size.
       with bl heap we need buffer size of 480, but that breaks audio at in cm. So we have,
       to use buffer size of 512 to get audio working but it makes stuttering as mentioned.
       so add support for a custom buffer size so we can use 480 buffer size.

Change-Id: Ib4b5222b4b64d42288e0ad5c95910e385d172374
---
 media/libstagefright/LPAPlayer.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/media/libstagefright/LPAPlayer.cpp b/media/libstagefright/LPAPlayer.cpp
index 9941313..dc93599 100755
--- a/media/libstagefright/LPAPlayer.cpp
+++ b/media/libstagefright/LPAPlayer.cpp
@@ -49,7 +49,10 @@
 
 static const char   mName[] = "LPAPlayer";
 
-#define MEM_BUFFER_SIZE 524288
+#ifndef LPA_DEFAULT_BUFFER_SIZE
+#define LPA_DEFAULT_BUFFER_SIZE 512
+#endif
+#define MEM_BUFFER_SIZE (LPA_DEFAULT_BUFFER_SIZE*1024)
 #define MEM_BUFFER_COUNT 2
 
 #define PCM_FORMAT 2
-- 
1.9.3

